Source/Scene/Scene.js

```diff
@@ -2696,13 +2750,166 @@ define([
         return object;
     };
 
+    var fragDepthRegex = /\bgl_FragDepthEXT\b/;
+    var discardRegex = /\bdiscard\b/;
+
+    function getDepthOnlyShaderProgram(scene, context, shaderProgram) {
+        var id = shaderProgram.id;
+        var cache = scene._depthOnlyShaderProgramCache;
+        var shader = cache[id];
+        if (!defined(shader)) {
+            var attributeLocations = shaderProgram._attributeLocations;
+            var fs = shaderProgram.fragmentShaderSource;
+
+            var writesFragDepth = false;
+            var discards = false;
+
+            var sources = fs.sources;
+            var length = sources.length;
+            for (var i = 0; i < length; ++i) {
+                if (fragDepthRegex.test(sources[i])) {
+                    writesFragDepth = true;
+                    break;
+                }
+                if (discardRegex.test(sources[i])) {
+                    discards = true;
+                    break;
+                }
+            }
+
+            if (!writesFragDepth && !discards) {
+                fs = new ShaderSource({
+                    sources : ['void main() { gl_FragColor = vec4(1.0); }']
+                });
+            }
+
+            shader = ShaderProgram.fromCache({
+                context : context,
+                vertexShaderSource : shaderProgram.vertexShaderSource,
+                fragmentShaderSource : fs,
+                attributeLocations : attributeLocations
+            });
+
+            cache[id] = shader;
```

Comment:
Is it a concern that this cache always grows and never shrinks?  Should it be possible for the original shader program to basically own this and free it when the ref count goes to zero?  Instead of being an object literal, this could be its own shader program cache instance if useful.

Also, is all this being shared with the shadow map pass code?